lib/azure/armrest/armrest_service.rb

```diff
@@ -215,23 +42,23 @@ def initialize(armrest_configuration, service_name, default_provider, options)
         set_service_api_version(options, service_name)
       end
 
-      # Returns a list of the available resource providers.
+      # Returns a list of the available resource providers. This is really
+      # just a wrapper for Azure::Armrest::Configuration#providers.
       #
-      def providers
-        url = url_with_api_version(configuration.api_version, @base_url, 'providers')
-        resp = rest_get(url)
-        JSON.parse(resp.body)["value"].map{ |hash| Azure::Armrest::ResourceProvider.new(hash) }
+      def list_providers
+        configuration.providers
       end
 
+      alias providers list_providers
```

Comment:
Yes, I'm adding aliases for now for backwards compatibility. We can remove them in a future release.